src/main/java/org/folio/domain/Condition.java

```diff
@@ -1,66 +1,62 @@
 
 import org.folio.rest.jaxrs.model.OpenLoan;
 import org.folio.rest.jaxrs.model.PatronBlockLimit;
 import org.folio.rest.jaxrs.model.UserSummary;
-import org.joda.time.DateTimeZone;
-import org.joda.time.Days;
-import org.joda.time.LocalDate;
 
 public enum Condition {
   // IDs come from resources\templates\db_scripts\populate-patron-block-conditions.sql
-
   MAX_NUMBER_OF_ITEMS_CHARGED_OUT("3d7c52dc-c732-4223-8bf8-e5917801386f",
-    (summary, limit) -> summary.getOpenLoans().size() >= limit.getValue()
+    (ctx) -> ctx.getUserSummary().getOpenLoans().size() >= ctx.getLimit().getValue()
   ),
 
   MAX_NUMBER_OF_LOST_ITEMS("72b67965-5b73-4840-bc0b-be8f3f6e047e",
-    (summary, limit) -> summary.getNumberOfLostItems() >= limit.getValue()
+    (ctx) -> ctx.getUserSummary().getNumberOfLostItems() >= ctx.getLimit().getValue()
   ),
 
   MAX_NUMBER_OF_OVERDUE_ITEMS("584fbd4f-6a34-4730-a6ca-73a6a6a9d845",
-    (summary, limit) -> summary.getOpenLoans().stream()
-      .filter(Condition::isLoanOverdue)
-      .count() >= limit.getValue()
+    (ctx) -> ctx.getUserSummary().getOpenLoans().stream()
+      .filter(openLoan -> ctx.getOverdueMinutes().get(openLoan.getLoanId()) > 0)
+      .count() >= ctx.getLimit().getValue()
   ),
 
   MAX_NUMBER_OF_OVERDUE_RECALLS("e5b45031-a202-4abb-917b-e1df9346fe2c",
-    (summary, limit) -> summary.getOpenLoans().stream()
-      .filter(Condition::isLoanOverdue)
+    (ctx) -> ctx.getUserSummary().getOpenLoans().stream()
+      .filter(openLoan -> ctx.getOverdueMinutes().get(openLoan.getLoanId()) > 0)
       .filter(OpenLoan::getRecall)
-      .count() >= limit.getValue()
+      .count() >= ctx.getLimit().getValue()
   ),
 
   RECALL_OVERDUE_BY_MAX_NUMBER_OF_DAYS("08530ac4-07f2-48e6-9dda-a97bc2bf7053",
-    (summary, limit) -> summary.getOpenLoans().stream()
-      .filter(Condition::isLoanOverdue)
+    (ctx) -> ctx.getUserSummary().getOpenLoans().stream()
+      .filter(openLoan -> ctx.getOverdueMinutes().get(openLoan.getLoanId()) > 0)
       .filter(OpenLoan::getRecall)
-      .map(Condition::getLoanOverdueDays)
-      .anyMatch(days -> days >= limit.getValue())
+      .map(openLoan -> getLoanOverdueDays(ctx.getOverdueMinutes().get(openLoan.getLoanId())))
+      .anyMatch(days -> days >= ctx.getLimit().getValue())
   ),
 
   MAX_OUTSTANDING_FEE_FINE_BALANCE("cf7a0d5f-a327-4ca1-aa9e-dc55ec006b8a",
-    (summary, limit) -> summary.getOutstandingFeeFineBalance()
-      .compareTo(BigDecimal.valueOf(limit.getValue())) >= 0
+    (ctx) -> ctx.getUserSummary().getOutstandingFeeFineBalance()
+      .compareTo(BigDecimal.valueOf(ctx.getLimit().getValue())) >= 0
   );
 
   private final String id;
-  private final BiPredicate<UserSummary, PatronBlockLimit> limitEvaluator;
+  private final Predicate<LimitEvaluationContext> limitEvaluator;
   private static final Map<String, Condition> idIndex = new HashMap<>(Condition.values().length);
+  private static final Double NUMBER_OF_MINUTES_IN_ONE_DAY = 1440.0;
```

Comment:
move this constant to OverduePeriodCalculatorService

Reply:
Done